shared-libs/transitions/src/transitions/index.js

```diff
@@ -96,60 +52,79 @@ const processChange = (change, callback) => {
         if (change.doc.transitions) {
           delete change.doc.transitions;
         }
-        module.exports.applyTransitions(change, () => {
-          processed++;
-          metadata
-            .update(change.seq)
-            .then(() => callback())
-            .catch(err => {
-              callback(err);
-            });
-        });
+        applyTransitions(change, callback);
       });
     })
     .catch(err => {
       logger.error('transitions: fetch failed for %s : %o', change.id, err);
-      return callback();
+      return callback(err);
     });
 };
 
-const deleteReadDocs = change => {
-  // we don't know if the deleted doc was a report or a message so
-  // attempt to delete both
-  const possibleReadDocIds = ['report', 'message'].map(
-    type => `read:${type}:${change.id}`
-  );
+// given a collection of docs this function will:
+// - deep clone the docs, to keep the original version of them safe
+// - add _id properties to the docs that are missing them
+// - hydrate the docs and generate info docs
+// - run loaded transitions over all docs
+// - returns docs, either updated or their original version
+const processDocs = docs => {
```

Comment:
I guess we could, but then everything would be done one doc at a time:
 - fetching lineage
 - creating the infodoc 

Perf wise, it would be much worse than bulking them.